src/qp/interface/qptransform.c

```diff
@@ -393,7 +393,7 @@ PetscErrorCode QPTEnforceEqByPenalty(QP qp, PetscReal rho_user, PetscBool rho_di
   /* newA = A + rho*BE'*BE */
   TRY( MatCreatePenalized(qp,rho,&newA) );
   TRY( QPSetOperator(child,newA) );
-  TRY( QPSetOperatorNullSpace(child,qp->R) );
+  //TODO what about nullspace
```

Comment:
Right A + rho*BE'*BE can have a different nullspace. I would just drop it.

Reply:
Yes, I think it could be derived but no one really cares at the moment, I guess. Do you think the TODO should be removed?

Reply:
I would just make the TODO clearer. Something along the lines of `derive null space of the new operator`?

Reply:
```suggestion
  //TODO Maybe we can derive the null space of newA?
```